es-core/src/resources/TextureData.cpp

```diff
@@ -46,33 +46,22 @@ bool TextureData::initSVGFromMemory(const unsigned char* fileData, size_t length
 
 	NSVGimage* svgImage = nsvgParse(copy, "px", DPI);
 	free(copy);
-	if (!svgImage)
+	if (!svgImage || (svgImage->width == 0) || (svgImage->height == 0))
 	{
 		LOG(LogError) << "Error parsing SVG image.";
 		return false;
 	}
 
 	// We want to rasterise this texture at a specific resolution. If the source size
 	// variables are set then use them otherwise set them from the parsed file
-	if ((mSourceWidth == 0.0f) && (mSourceHeight == 0.0f))
-	{
-		mSourceWidth = svgImage->width;
+	if (mSourceHeight == 0.0f)
```

Comment:
Why have this if?

Reply:
To make sure that mSourceHeight is not 0, width will always be based on the height.

Reply:
Ah, my question is: why not just always set it instead of running the if. :)

Reply:
Because mSourceHeight is our wanted size, which probably is not what is stored as the default height in the svg.
This is just a fallback if an mSourceHeight of 0.0f was requested.